ui-core/src/components/inputs/InputWithSuggestions/InputWithSuggestions.tsx

```diff
@@ -0,0 +1,174 @@
+  const [filteredSuggestions, setFilteredSuggestions] = useState<T[]>([]);
+  const [activeSuggestionIndex, setActiveSuggestionIndex] = useState(-1);
+  const [inputValue, setInputValue] = useState(value ? getSuggestionLabel(value) : '');
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const showSuggestions = filteredSuggestions.length > 0;
+
+  const focusInput = () => inputRef.current?.focus();
+
+  const clearInput = () => {
+    setInputValue('');
+    inputRef.current?.focus();
+  };
+
+  const icons = [
+    ...(inputValue
+      ? [
+          {
+            icon: <X size={small ? 'sm' : 'lg'} />,
+            action: clearInput,
+            className: 'clear-icon',
+          },
+        ]
+      : []), // Conditionally include the clear icon only when inputValue is not empty
+    ...(suggestions.length > 0
+      ? [
+          {
+            icon: <ChevronDown size={small ? 'sm' : 'lg'} />,
+            action: focusInput,
+            className: 'chevron-icon',
+          },
+        ]
+      : []), // Conditionally include the chevron icon only when suggestions are not empty
+  ];
+
+  useEffect(() => {
+    if (value) {
+      setInputValue(getSuggestionLabel(value));
+    }
+  }, [value, getSuggestionLabel]);
+
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    const userInput = normalizeString(e.currentTarget.value).trim();
+    setInputValue(e.currentTarget.value);
+
+    if (userInput.trim() === '') {
+      setFilteredSuggestions([]);
+      return;
+    }
+
+    const filtered = suggestions.filter((suggestion) => {
+      const suggestionLabel = normalizeString(getSuggestionLabel(suggestion).toLowerCase());
+      return exactSearch
+        ? suggestionLabel.startsWith(userInput.toLowerCase())
+        : suggestionLabel.includes(userInput.toLowerCase());
+    });
+    setFilteredSuggestions(filtered);
+  };
+
+  const handleKeyDown: KeyboardEventHandler<HTMLInputElement> = (e) => {
+    if (e.key === 'ArrowDown') {
+      setActiveSuggestionIndex((prev) => (prev < filteredSuggestions.length - 1 ? prev + 1 : prev));
+    } else if (e.key === 'ArrowUp') {
+      setActiveSuggestionIndex((prev) => (prev > 0 ? prev - 1 : prev));
+    } else if ((e.key === 'Enter' || e.key === 'Tab') && activeSuggestionIndex >= 0) {
+      selectSuggestion(activeSuggestionIndex);
+    } else if (e.key === 'Escape') {
+      setFilteredSuggestions([]);
+    }
+  };
+
+  const selectSuggestion = (index: number) => {
+    const selectedSuggestion = filteredSuggestions[index];
+    setInputValue(getSuggestionLabel(selectedSuggestion));
+    onChange(selectedSuggestion);
+    setFilteredSuggestions([]);
+  };
+
+  const handleInputFocus: FocusEventHandler<HTMLInputElement> = (e) => {
+    e.stopPropagation();
+    if (inputRef.current && inputRef.current.value === '') {
+      const sortedSuggestions = suggestions.sort((a, b) =>
+        getSuggestionLabel(a).localeCompare(getSuggestionLabel(b))
+      );
```

Comment:
If you sort in this function, we will sort the suggestions each time this function is launched.

Maybe we could just put the sorted list in a useMemo to do it one it when we get the suggestions ? 

```suggestion
  const sortedSuggestions = useMemo(
    () => suggestions.sort((a, b) => getSuggestionLabel(a).localeCompare(getSuggestionLabel(b))),
    [suggestions]
  );
```

Reply:
Fixed :)